xqd_request.go

```diff
@@ -152,6 +152,33 @@ func (i *Instance) xqd_req_header_names_get(handle int32, addr int32, maxlen int
 	return xqd_multivalue(i.memory, names, addr, maxlen, cursor, ending_cursor_out, nwritten_out)
 }
 
+func (i *Instance) xqd_req_header_value_get(handle int32, name_addr int32, name_size int32, addr int32, maxlen int32, nwritten_out int32) int32 {
+	var r = i.requests.Get(int(handle))
+	if r == nil {
+		return XqdErrInvalidHandle
+	}
+
+	var buf = make([]byte, name_size)
+	var _, err = i.memory.ReadAt(buf, int64(name_addr))
+	if err != nil {
+		return XqdError
+	}
+
+	var header = http.CanonicalHeaderKey(string(buf))
+
+	i.abilog.Printf("req_header_value_get: handle=%d header=%q\n", handle, header)
+
+	var values, ok = r.Header[header]
```

Comment:
`r.Header.Get(header)` will return either the first value of that header or the empty string. Since we're not doing anything special when there's zero or more than one header value, we could probably reduce this.

On a related note, `.Get()` will canonicalize the header also, so it could just be `.Get(string(buf))`

Reply:
Oh, great! I didn't pay attention to `Header.Get()`. Way simpler indeed.

I'm going to update that PR.